I would even go more concrete to use
```
depends_on('openblas threads=openmp', when='+openmp ^openblas')
```
(Works well with clingo, does fail with the original dependency solver)

> I haven't tested USE_LOCKING for openblas, but I will soon.

Wuhu, thanks to the great test/check implementation in the `blaspp` package, I gave it a try already: `spack install --test root blaspp`. This is the issue I saw:
```
217 tests FAILED for batch-gemm
...
BLAS : Bad memory unallocation!
```
Indeed, enabling `USE_LOCKING=1` introduced with OpenBLAS 0.3.7+ solves the test issues! :tada:

I updated this PR to reflect the changes in OpenBLAS and BLAS++ and will wait for your and @mgates3's confirmation that this indeed is the solution.

I already did `spack install --test root blaspp` tests with OpenBLAS (Ubuntu 20.04/x86_64, GCC 10.2, original concretizer):

### 0.3.6

- [x] `spack install --test root blaspp ^openblas@0.3.6`
Spack abort by default with conflict (original concretizer).
`USE_LOCKING` came in [OpenBLAS 0.3.7 according to their docs](https://github.com/xianyi/OpenBLAS/blob/v0.3.7/Makefile.rule#L61-L65), so I added a new specific conflict

- [x] `spack install --test root blaspp ^openblas@0.3.6 threads=openmp`
BLAS++ 2020.10.02 (latest tag) CMake issue:
```
     30       BLAS library not found.
  >> 31    CMake Error at CMakeLists.txt:295 (message):
     32      BLAS++ requires a BLAS library and none was found.  Ensure that it is
     33      accessible in environment variables $CPATH, $LIBRARY_PATH, and
     34      $LD_LIBRARY_PATH.
```

### 0.3.7

- [x] `spack install --test root blaspp ^openblas@0.3.7`
same CMake error as with OpenBLAS 0.3.6

Otherwise this _should_ work because of the `+locking` default that came in this OpenBLAS release.

- [x] `spack install --test root blaspp ^openblas@0.3.7 threads=openmp`
same CMake error

- [ ] Do we want to simply add a conflict for selected old OpenBLAS releases in BLAS++'s `package.py`? Maybe that's also solved with newer BLAS++ releases that reworked the CMake, but the nwest release is still 2020.10.02 (latest tag) @mgates3

### 0.3.8

- [x] `spack install --test root blaspp ^openblas@0.3.8`
works

### 0.3.9

- [x] `spack install --test root blaspp ^openblas@0.3.9`
works

### 0.3.13 (latest)

Probably the most important test, because this is the *default* user experience with Spack:

- [x] `spack install --test root blaspp`
works

Just checking, because yolo:

- [x] `spack install --test root blaspp ^openblas threads=openmp`
works

### Any without locking

- [x] `spack install --test root blaspp ^openblas ~locking`
conflict added for all OpenBLAS releases >=0.3.7 (see 0.3.6 note above)

--- a/var/spack/repos/builtin/packages/blaspp/package.py
+++ b/var/spack/repos/builtin/packages/blaspp/package.py
@@ -29,8 +29,13 @@ class Blaspp(CMakePackage, CudaPackage):
     depends_on('cmake@3.15.0:', type='build')
     depends_on('blas')
 
+    # only supported with clingo solver: virtual dependency preferences
+    # depends_on('openblas threads=openmp', when='+openmp ^openblas')
+
     # BLASpp tests will fail when using openblas > 0.3.5 without multithreading support
-    conflicts('^openblas@0.3.6: threads=none', msg='BLASpp requires openblas multithreading support')
+    # locking is only supported in openblas 3.7+
+    conflicts('^openblas@0.3.6 threads=none', msg='BLASpp requires a threadsafe openblas')
+    conflicts('^openblas@0.3.7: ~locking', msg='BLASpp requires a threadsafe openblas')
 
     def cmake_args(self):
         spec = self.spec

